client/question/components/LearnerViewDirective.js

```diff
@@ -661,61 +661,93 @@ tie.directive('learnerView', [function() {
-          var hiddenAttributeName = determineHiddenAttributeNameForBrowser();
-          if (hiddenAttributeName !== null &&
-            document[determineHiddenAttributeNameForBrowser()]) {
-            EventHandlerService.createSessionPauseEvent(
-              SessionIdService.getSessionId());
-          } else {
-            EventHandlerService.createSessionResumeEvent(
-              SessionIdService.getSessionId());
+          var hiddenAttributeName = (
+            $scope.determineHiddenAttributeNameForBrowser());
+          if (hiddenAttributeName !== null) {
+            if ($scope.isDocumentHidden(hiddenAttributeName)) {
+              EventHandlerService.createSessionPauseEvent(
+                SessionIdService.getSessionId());
+            } else {
+              EventHandlerService.createSessionResumeEvent(
+                SessionIdService.getSessionId());
+            }
           }
         };
 
+        // Move document[hiddenAttributeName] getter into function for testing.
+        $scope.isDocumentHidden = function(hiddenAttributeName) {
+          return document[hiddenAttributeName];
+        };
+
+        // Move document.hidden getter into function for testing.
+        $scope.getHiddenAttribute = function() {
+          return document.hidden;
+        };
+
+        // Move document.msHidden getter into function for testing.
+        $scope.getMsHiddenAttribute = function() {
+          return document.msHidden;
+        };
+
+        // Move document.webkitHidden getter into function for testing.
+        $scope.getWebkitHiddenAttribute = function() {
+          return document.webkitHidden;
+        };
+
         /**
          * Different browsers call the "hidden" attribute different things.
          * This method determines what the current browser calls its "hidden"
          * attribute and returns it.
          */
-        var determineHiddenAttributeNameForBrowser = function() {
-          if (typeof document.hidden !== 'undefined') {
+        $scope.determineHiddenAttributeNameForBrowser = function() {
+          if (typeof $scope.getHiddenAttribute() !== 'undefined') {
             // Opera 12.10 and Firefox 18 and later support
             return 'hidden';
-          } else if (typeof document.msHidden !== 'undefined') {
+          } else if (typeof $scope.getMsHiddenAttribute() !== 'undefined') {
             return 'msHidden';
-          } else if (typeof document.webkitHidden !== 'undefined') {
+          } else if (typeof $scope.getWebkitHiddenAttribute() !== 'undefined') {
             return 'webkitHidden';
           }
           return null;
         };
 
-        var setEventListenerForVisibilityChange = function() {
-          var hiddenAttributeName = determineHiddenAttributeNameForBrowser();
+        $scope.determineVisibilityChangeAttributeNameForBrowser = function() {
+          // Handle page visibility change
+          if (typeof $scope.getHiddenAttribute() !== 'undefined') {
```

Comment:
There's duplication here and above -- probably cleaner to have a function that gets the browser type and then just read off the variable name (or in fact the attr prefix) from an object mapping browser type to attr prefix.